src/main/java/com/example/springbootbookshop/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,58 @@
 package com.example.springbootbookshop.service.impl;
 
+import com.example.springbootbookshop.dto.BookDto;
+import com.example.springbootbookshop.dto.CreateBookRequestDto;
 import com.example.springbootbookshop.entity.Book;
+import com.example.springbootbookshop.mapper.BookMapper;
 import com.example.springbootbookshop.repository.BookRepository;
 import com.example.springbootbookshop.service.BookService;
+import jakarta.persistence.EntityNotFoundException;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 
+@RequiredArgsConstructor
 @Component
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public Book save(CreateBookRequestDto book) {
+        return bookRepository.save(bookMapper.toBook(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException("Can`t find"
+                        + "book with id:" + id));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto update(CreateBookRequestDto bookRequestDto, Long id) {
+        Optional<Book> bookOptional = bookRepository.findById(id);
+        if (bookOptional.isPresent()) {
+            Book book = bookMapper.toBook(bookRequestDto);
+            book.setId(id);
+            bookRepository.save(book);
+            return bookMapper.toDto(book);
+        } else {
+            throw new EntityNotFoundException("Book with ID " + id + " not found");
+        }
```

Comment:
let's improve this a bit :)

first thing is that you can leverage Mapstruct to perform the update:
```
void updateBook(CreateBookRequestDto dto, @MappingTarget Book book);
```

then:
```
Book bookToUpdate = bookRepository.findById(id).orElseThrow(...);
bookMapper.updateBook(dto, bookToUpdate);
bookRepository.save(book);
return bookMapper.toDto(book);
```